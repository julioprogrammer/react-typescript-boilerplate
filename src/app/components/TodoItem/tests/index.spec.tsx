import * as React from 'react';
import { Provider } from 'react-redux';
import { render } from 'enzyme';
import configureStore from 'redux-mock-store';
import { TodoModel } from 'app/models';
import { TodoItem } from '../index';

const todoMock : TodoModel = {
  id: 1,
  text: 'Teste component render',
  completed: false
};

const mockStore = configureStore();

const store = mockStore(todoMock);

describe('TodoItem', () => {
  describe('Render', () => {
    it('should match snapshot', () => {
      const rendered =  render(
        <Provider store={store}>
          <TodoItem todo={todoMock} />
        </Provider>
      );
      expect(rendered).toMatchSnapshot();
    })
  });
});
