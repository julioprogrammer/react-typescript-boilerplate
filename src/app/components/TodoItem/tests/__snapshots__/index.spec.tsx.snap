// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Render should match snapshot 1`] = `
<li
  class=""
>
  <div
    class="view"
  >
    <input
      class="toggle"
      type="checkbox"
    />
    <label>
      Teste component render
    </label>
    <button
      class="destroy"
    />
  </div>
</li>
`;
